app/Http/Controllers/Admin/BookController.php

```diff
@@ -52,53 +50,53 @@ public function store(BookPostRequest $request)
         $book->slug = $str;
 
         if ($book->save()) {
-            return redirect("/admin/book/add")->with('success', 'Kayıt Başarılı!');
+            Session::flash('bookRegistrationSuccessful', 'Kitap Kaydı Başarılı!');
         } else {
-            return redirect("/admin/book/add")->with('error', 'Kayıt Başarısız!');;
+            Session::flash('bookRegistrationFailed', 'Kitap Kaydı Başarısız!');
         }
+        return redirect()->action([BookController::class, 'index']);
     }
     //FormRequest
-    public function edit(Request $request, $id)
+    public function edit(Request $request, Book $book)
     {
-        $oldBook = Book::where("id", $id)->firstOrFail();
-        $filePath = $oldBook->image;
-        $str = Str::slug($request->name, '-');
-
-        if ($request->file("image") != null) {
-            if ($oldBook->image != "no-image/no-image.jpeg") {
-                Storage::disk('uploads')->delete($oldBook->image);
-            }
-            $filePath = Storage::disk('uploads')->put('books', $request->file("image"), 'public');
+        $oldbook = Book::where("id", $book->id)->firstOrFail();
+        if ($request->hasFile('image')) {
+            $filePath = Storage::disk('storage')->put('books', $request->file("image"), 'public');
+        } else {
+            $filePath = $oldbook->image;
         }
-
-        $book = Book::where("id", $id)->update([
+        $str = Str::slug($request->name, '-');
+        $bookUpdate = Book::where("id", $book->id)->update([
             "name" => $request->input('name'),
             "author_id" => $request->input('author_id'),
             "book_no" => $request->input('book_no'),
             "status" => $request->input('status'),
             "image" => $filePath,
-            "slug" => $str,
+            "slug" => $str
         ]);
-
-        if ($book) {
-            return redirect("/admin/books/list")->with('successUpdate', 'Güncelleme Başarılı!');
+        if ($bookUpdate) {
+            if ($oldbook->image != "no-image/no-image.jpeg") {
```

Comment:
1- sadece image için book datasının tamamını sırf bu iş için değişkende tutmak yerine $oldBookImage şeklinde bir değişkene değeri tanımlayıp kullanman daha pratik olur.
2- https://github.com/BenSampo/laravel-enum "no-image/no-image.jpeg" şu pathi her yerde kullanıyorsun. laravel-enum paketini kurup ImageEnum classı falan oluşturup içine NO_IMAGE_PATH şeklinde bir enum tanımlayıp kullanman daha doğru olur. bu pathi her yerde statik olarak kullanıyorsun. Bu tarz tekrar eden statik içerikleri enum, const vs. olarak ilgili yerlere tanımlayıp çağırarak kullanmak daha doğru bir kullanım olur. const değerini güncellediğinde her yerde otomatik olarak güncellenebilir olur. Yoksa pathi güncellemek istediğimizde CTR+SHIFT+F yapıp global search replace ile yapmak pratik değil.